src/Concepts.md

```diff
@@ -0,0 +1,57 @@
+# Concepts
+
+Embedded Controllers are the unsung heroes of the modern laptop, quietly handling power management, thermal control, 
+battery charging, lid sensors, keyboard scan matrices, and sometimes even security functions. 
+There's a surprising amount of complexity tucked away in that little chip.
+
+The drivers and handlers responsible for managing these subsystems must be secure, reliable, and easy to adopt with confidence. 
+This calls for a standardized, community-moderated approach—one that still leaves room for innovation and platform-specific 
+differentiation.
+
+The Open Device Partnership defines a pattern for implementing this architecture in Rust, typically built on top of the [Embassy](https://embassy.dev/) 
+framework for embedded systems. Embassy provides key building blocks like Hardware Abstraction Layers (HALs), consistent timing 
+models, and support for both asynchronous and blocking execution modes.
+
+Hardware components issue events or respond to signals transmitted over data buses such as eSPI, SMBus, or I²C. These signals 
+are monitored or driven by software, forming the basis for orchestrating and governing hardware behavior
+
+Historically, much of this firmware has been vendor-supplied and tightly coupled to specific EC hardware. It's often written 
+in C or even assembly, and may be vulnerable to memory-unsafe operations or unintended behavior introduced by seemingly 
+harmless changes.
+
+As computing devices grow more complex and user data becomes increasingly sensitive, the need for provable safety 
```

Comment:
It might be good to mention that AI brings a whole new level of device security requirements to consider: protecting AI models from extraction or nefarious use, protecting the private data that feeds AI inference, etc.  Basically device security has always been important but with AI scenarios, it's even more important.